The job **`x86_64-gnu-nopt`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/6290367721/job/17077714759) [(plain)](https://github.com/rust-lang-ci/rust/commit/74d6a4590419981c14f8a063f6d2ba8d564da521/checks/17077714759/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [assembly] tests/assembly/closure-inherit-target-feature.rs stdout ----

error: verification with 'FileCheck' failed
status: exit status: 1
command: "/checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/closure-inherit-target-feature/closure-inherit-target-feature.s" "/checkout/tests/assembly/closure-inherit-target-feature.rs" "--allow-unused-prefixes" "--check-prefixes" "CHECK,NONMSVC" "--dump-input-context" "100"
--- stderr -------------------------------
/checkout/tests/assembly/closure-inherit-target-feature.rs:31:12: error: CHECK: expected string not found in input
/checkout/tests/assembly/closure-inherit-target-feature.rs:31:12: error: CHECK: expected string not found in input
 // CHECK: blendps
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/closure-inherit-target-feature/closure-inherit-target-feature.s:107:109: note: scanning from here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/closure-inherit-target-feature/closure-inherit-target-feature.s:107:109: note: scanning from here
_ZN30closure_inherit_target_feature20sse41_blend_noinline28_$u7b$$u7b$closure$u7d$$u7d$17h8e28e2f3c9248f9eE:
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/closure-inherit-target-feature/closure-inherit-target-feature.s:119:58: note: possible intended match here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/closure-inherit-target-feature/closure-inherit-target-feature.s:119:58: note: possible intended match here
 call qword ptr [rip + _ZN4core9core_arch3x865sse4112_mm_blend_ps17h14d90f400076c061E@GOTPCREL]
/checkout/tests/assembly/closure-inherit-target-feature.rs:48:12: error: CHECK: expected string not found in input
/checkout/tests/assembly/closure-inherit-target-feature.rs:48:12: error: CHECK: expected string not found in input
 // CHECK: blendps
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/closure-inherit-target-feature/closure-inherit-target-feature.s:132:22: note: scanning from here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/closure-inherit-target-feature/closure-inherit-target-feature.s:132:22: note: scanning from here
sse41_blend_doinline:
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/closure-inherit-target-feature/closure-inherit-target-feature.s:149:50: note: possible intended match here
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/closure-inherit-target-feature/closure-inherit-target-feature.s:149:50: note: possible intended match here
 call _ZN30closure_inherit_target_feature20sse41_blend_doinline28_$u7b$$u7b$closure$u7d$$u7d$17h7b11557ec6cc5ec8E

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/closure-inherit-target-feature/closure-inherit-target-feature.s
Check file: /checkout/tests/assembly/closure-inherit-target-feature.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
            .
            .
            .
            .
            7:  .type _ZN4core9core_arch3x865sse4112_mm_blend_ps17h14d90f400076c061E,@function 
            8: _ZN4core9core_arch3x865sse4112_mm_blend_ps17h14d90f400076c061E: 
            9:  .cfi_startproc 
           10:  mov rax, rdi 
           11:  movaps xmm1, xmmword ptr [rsi] 
           12:  movaps xmm0, xmmword ptr [rdx] 
           13:  blendps xmm0, xmm1, 10 
           14:  movaps xmmword ptr [rdi], xmm0 
           15:  ret 
           16: .Lfunc_end0: 
           17:  .size _ZN4core9core_arch3x865sse4112_mm_blend_ps17h14d90f400076c061E, .Lfunc_end0-_ZN4core9core_arch3x865sse4112_mm_blend_ps17h14d90f400076c061E 
           18:  .cfi_endproc 
           19:  
           20:  .section .text.sse41_blend_nofeature,"ax",@progbits 
           21:  .globl sse41_blend_nofeature 
           22:  .p2align 4, 0x90 
           23:  .type sse41_blend_nofeature,@function 
           24: sse41_blend_nofeature: 
           25:  .cfi_startproc 
           26:  sub rsp, 88 
           27:  .cfi_def_cfa_offset 96 
           28:  mov rax, rdi 
           29:  mov qword ptr [rsp], rax 
           30:  movaps xmm0, xmmword ptr [rsi] 
           31:  movaps xmmword ptr [rsp + 16], xmm0 
           32:  movaps xmm0, xmmword ptr [rdx] 
           33:  movaps xmmword ptr [rsp + 32], xmm0 
           34:  movaps xmm0, xmmword ptr [rsp + 16] 
           35:  movaps xmmword ptr [rsp + 48], xmm0 
           36:  movaps xmm0, xmmword ptr [rsp + 32] 
           37:  movaps xmmword ptr [rsp + 64], xmm0 
           38:  lea rsi, [rsp + 15] 
           39:  lea rdx, [rsp + 48] 
           40:  lea rcx, [rsp + 64] 
           41:  call _ZN30closure_inherit_target_feature21sse41_blend_nofeature28_$u7b$$u7b$closure$u7d$$u7d$17hcd5f540674891f77E 
           42:  mov rax, qword ptr [rsp] 
           43:  add rsp, 88 
           44:  .cfi_def_cfa_offset 8 
           45:  ret 
           46: .Lfunc_end1: 
           47:  .size sse41_blend_nofeature, .Lfunc_end1-sse41_blend_nofeature 
           48:  .cfi_endproc 
           49:  
           50:  .section ".text._ZN30closure_inherit_target_feature21sse41_blend_nofeature28_$u7b$$u7b$closure$u7d$$u7d$17hcd5f540674891f77E","ax",@progbits 
           51:  .p2align 4, 0x90 
           52:  .type _ZN30closure_inherit_target_feature21sse41_blend_nofeature28_$u7b$$u7b$closure$u7d$$u7d$17hcd5f540674891f77E,@function 
           53: _ZN30closure_inherit_target_feature21sse41_blend_nofeature28_$u7b$$u7b$closure$u7d$$u7d$17hcd5f540674891f77E: 
           54:  .cfi_startproc 
           55:  sub rsp, 56 
           56:  .cfi_def_cfa_offset 64 
           57:  mov rax, rdi 
           58:  mov qword ptr [rsp + 8], rax 
           59:  movaps xmm0, xmmword ptr [rdx] 
           60:  movaps xmmword ptr [rsp + 16], xmm0 
           61:  movaps xmm0, xmmword ptr [rcx] 
           62:  movaps xmmword ptr [rsp + 32], xmm0 
           63:  lea rsi, [rsp + 16] 
           64:  lea rdx, [rsp + 32] 
           65:  call qword ptr [rip + _ZN4core9core_arch3x865sse4112_mm_blend_ps17h14d90f400076c061E@GOTPCREL] 
           66:  mov rax, qword ptr [rsp + 8] 
           67:  add rsp, 56 
           68:  .cfi_def_cfa_offset 8 
           69:  ret 
           70: .Lfunc_end2: 
           71:  .size _ZN30closure_inherit_target_feature21sse41_blend_nofeature28_$u7b$$u7b$closure$u7d$$u7d$17hcd5f540674891f77E, .Lfunc_end2-_ZN30closure_inherit_target_feature21sse41_blend_nofeature28_$u7b$$u7b$closure$u7d$$u7d$17hcd5f540674891f77E 
           72:  .cfi_endproc 
           73:  
           74:  .section .text._ZN30closure_inherit_target_feature20sse41_blend_noinline17hba8778d3197b1ddaE,"ax",@progbits 
           75:  .globl _ZN30closure_inherit_target_feature20sse41_blend_noinline17hba8778d3197b1ddaE 
           76:  .p2align 4, 0x90 
           77:  .type _ZN30closure_inherit_target_feature20sse41_blend_noinline17hba8778d3197b1ddaE,@function 
           78: _ZN30closure_inherit_target_feature20sse41_blend_noinline17hba8778d3197b1ddaE: 
           79:  .cfi_startproc 
           80:  sub rsp, 88 
           81:  .cfi_def_cfa_offset 96 
           82:  mov rax, rdi 
           83:  mov qword ptr [rsp], rax 
           84:  movaps xmm0, xmmword ptr [rsi] 
           85:  movaps xmmword ptr [rsp + 16], xmm0 
           86:  movaps xmm0, xmmword ptr [rdx] 
           87:  movaps xmmword ptr [rsp + 32], xmm0 
           88:  movaps xmm0, xmmword ptr [rsp + 16] 
           89:  movaps xmmword ptr [rsp + 48], xmm0 
           90:  movaps xmm0, xmmword ptr [rsp + 32] 
           91:  movaps xmmword ptr [rsp + 64], xmm0 
           92:  lea rsi, [rsp + 15] 
           93:  lea rdx, [rsp + 48] 
           94:  lea rcx, [rsp + 64] 
           95:  call _ZN30closure_inherit_target_feature20sse41_blend_noinline28_$u7b$$u7b$closure$u7d$$u7d$17h8e28e2f3c9248f9eE 
           96:  mov rax, qword ptr [rsp] 
           97:  add rsp, 88 
           98:  .cfi_def_cfa_offset 8 
           99:  ret 
          100: .Lfunc_end3: 
          101:  .size _ZN30closure_inherit_target_feature20sse41_blend_noinline17hba8778d3197b1ddaE, .Lfunc_end3-_ZN30closure_inherit_target_feature20sse41_blend_noinline17hba8778d3197b1ddaE 
          102:  .cfi_endproc 
          103:  
          104:  .section ".text._ZN30closure_inherit_target_feature20sse41_blend_noinline28_$u7b$$u7b$closure$u7d$$u7d$17h8e28e2f3c9248f9eE","ax",@progbits 
          105:  .p2align 4, 0x90 
          106:  .type _ZN30closure_inherit_target_feature20sse41_blend_noinline28_$u7b$$u7b$closure$u7d$$u7d$17h8e28e2f3c9248f9eE,@function 
          107: _ZN30closure_inherit_target_feature20sse41_blend_noinline28_$u7b$$u7b$closure$u7d$$u7d$17h8e28e2f3c9248f9eE: 
check:31'0                                                                                                                 X error: no match found
          108:  .cfi_startproc 
check:31'0     ~~~~~~~~~~~~~~~~
          109:  sub rsp, 56 
check:31'0     ~~~~~~~~~~~~~
          110:  .cfi_def_cfa_offset 64 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          111:  mov rax, rdi 
check:31'0     ~~~~~~~~~~~~~~
          112:  mov qword ptr [rsp + 8], rax 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          113:  movaps xmm0, xmmword ptr [rdx] 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          114:  movaps xmmword ptr [rsp + 16], xmm0 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          115:  movaps xmm0, xmmword ptr [rcx] 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          116:  movaps xmmword ptr [rsp + 32], xmm0 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          117:  lea rsi, [rsp + 16] 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~
          118:  lea rdx, [rsp + 32] 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~
          119:  call qword ptr [rip + _ZN4core9core_arch3x865sse4112_mm_blend_ps17h14d90f400076c061E@GOTPCREL] 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:31'1                                                              ?                                       possible intended match
          120:  mov rax, qword ptr [rsp + 8] 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          121:  add rsp, 56 
check:31'0     ~~~~~~~~~~~~~
          122:  .cfi_def_cfa_offset 8 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~
          123:  ret 
check:31'0     ~~~~~
          124: .Lfunc_end4: 
check:31'0     ~~~~~~~~~~~~~
          125:  .size _ZN30closure_inherit_target_feature20sse41_blend_noinline28_$u7b$$u7b$closure$u7d$$u7d$17h8e28e2f3c9248f9eE, .Lfunc_end4-_ZN30closure_inherit_target_feature20sse41_blend_noinline28_$u7b$$u7b$closure$u7d$$u7d$17h8e28e2f3c9248f9eE 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          126:  .cfi_endproc 
check:31'0     ~~~~~~~~~~~~~~
          127:  
check:31'0     ~
          128:  .section .text.sse41_blend_doinline,"ax",@progbits 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          129:  .globl sse41_blend_doinline 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          130:  .p2align 4, 0x90 
check:31'0     ~~~~~~~~~~~~~~~~~~
          131:  .type sse41_blend_doinline,@function 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          132: sse41_blend_doinline: 
check:31'0     ~~~~~~~~~~~~~~~~~~~~~
check:48'0                          X error: no match found
          133:  .cfi_startproc 
check:48'0     ~~~~~~~~~~~~~~~~
          134:  sub rsp, 88 
check:48'0     ~~~~~~~~~~~~~
          135:  .cfi_def_cfa_offset 96 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          136:  mov rax, rdi 
check:48'0     ~~~~~~~~~~~~~~
          137:  mov qword ptr [rsp], rax 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          138:  movaps xmm0, xmmword ptr [rsi] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          139:  movaps xmmword ptr [rsp + 16], xmm0 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          140:  movaps xmm0, xmmword ptr [rdx] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          141:  movaps xmmword ptr [rsp + 32], xmm0 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          142:  movaps xmm0, xmmword ptr [rsp + 16] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          143:  movaps xmmword ptr [rsp + 48], xmm0 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          144:  movaps xmm0, xmmword ptr [rsp + 32] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          145:  movaps xmmword ptr [rsp + 64], xmm0 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          146:  lea rsi, [rsp + 15] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~
          147:  lea rdx, [rsp + 48] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~
          148:  lea rcx, [rsp + 64] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~
          149:  call _ZN30closure_inherit_target_feature20sse41_blend_doinline28_$u7b$$u7b$closure$u7d$$u7d$17h7b11557ec6cc5ec8E 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:48'1                                                      ?                                                                 possible intended match
          150:  mov rax, qword ptr [rsp] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~
          151:  add rsp, 88 
check:48'0     ~~~~~~~~~~~~~
          152:  .cfi_def_cfa_offset 8 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~
          153:  ret 
check:48'0     ~~~~~
          154: .Lfunc_end5: 
check:48'0     ~~~~~~~~~~~~~
          155:  .size sse41_blend_doinline, .Lfunc_end5-sse41_blend_doinline 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          156:  .cfi_endproc 
check:48'0     ~~~~~~~~~~~~~~
          157:  
check:48'0     ~
          158:  .section ".text._ZN30closure_inherit_target_feature20sse41_blend_doinline28_$u7b$$u7b$closure$u7d$$u7d$17h7b11557ec6cc5ec8E","ax",@progbits 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          159:  .p2align 4, 0x90 
check:48'0     ~~~~~~~~~~~~~~~~~~
          160:  .type _ZN30closure_inherit_target_feature20sse41_blend_doinline28_$u7b$$u7b$closure$u7d$$u7d$17h7b11557ec6cc5ec8E,@function 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          161: _ZN30closure_inherit_target_feature20sse41_blend_doinline28_$u7b$$u7b$closure$u7d$$u7d$17h7b11557ec6cc5ec8E: 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          162:  .cfi_startproc 
check:48'0     ~~~~~~~~~~~~~~~~
          163:  sub rsp, 56 
check:48'0     ~~~~~~~~~~~~~
          164:  .cfi_def_cfa_offset 64 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~
          165:  mov rax, rdi 
check:48'0     ~~~~~~~~~~~~~~
          166:  mov qword ptr [rsp + 8], rax 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          167:  movaps xmm0, xmmword ptr [rdx] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          168:  movaps xmmword ptr [rsp + 16], xmm0 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          169:  movaps xmm0, xmmword ptr [rcx] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          170:  movaps xmmword ptr [rsp + 32], xmm0 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          171:  lea rsi, [rsp + 16] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~
          172:  lea rdx, [rsp + 32] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~
          173:  call qword ptr [rip + _ZN4core9core_arch3x865sse4112_mm_blend_ps17h14d90f400076c061E@GOTPCREL] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          174:  mov rax, qword ptr [rsp + 8] 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          175:  add rsp, 56 
check:48'0     ~~~~~~~~~~~~~
          176:  .cfi_def_cfa_offset 8 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~
          177:  ret 
check:48'0     ~~~~~
          178: .Lfunc_end6: 
check:48'0     ~~~~~~~~~~~~~
          179:  .size _ZN30closure_inherit_target_feature20sse41_blend_doinline28_$u7b$$u7b$closure$u7d$$u7d$17h7b11557ec6cc5ec8E, .Lfunc_end6-_ZN30closure_inherit_target_feature20sse41_blend_doinline28_$u7b$$u7b$closure$u7d$$u7d$17h7b11557ec6cc5ec8E 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          180:  .cfi_endproc 
check:48'0     ~~~~~~~~~~~~~~
          181:  
check:48'0     ~
          182:  .ident "rustc version 1.74.0-nightly (74d6a4590 2023-09-24)" 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
          183:  .section ".note.GNU-stack","",@progbits 
check:48'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
------------------------------------------



```

</details>

--- a/tests/assembly/closure-inherit-target-feature.rs
+++ b/tests/assembly/closure-inherit-target-feature.rs
@@ -1,0 +1,58 @@
+// only-x86_64
+// assembly-output: emit-asm
+// make sure the feature is not enabled at compile-time
+// compile-flags: -C opt-level=3 -C target-feature=-sse4.1 -C llvm-args=-x86-asm-syntax=intel
+
+#![feature(target_feature_11)]
+#![crate_type = "rlib"]
+
+use std::arch::x86_64::{__m128, _mm_blend_ps};
+
+#[no_mangle]
+pub unsafe fn sse41_blend_nofeature(x: __m128, y: __m128) -> __m128 {
+    let f = {
+        // check that _mm_blend_ps is not being inlined into the closure
+        // CHECK-LABEL: {{sse41_blend_nofeature.*closure.*:}}
+        // CHECK-NOT: blendps
+        // CHECK: {{call .*_mm_blend_ps.*}}
+        // CHECK-NOT: blendps
+        // CHECK: ret
+        #[inline(never)] |x, y| _mm_blend_ps(x, y, 0b0101)
+    };
+    f(x, y)
+}
+
+#[target_feature(enable = "sse4.1")]
+pub fn sse41_blend_noinline(x: __m128, y: __m128) -> __m128 {
+    let f = {
+        // check that _mm_blend_ps is being inlined into the closure
+        // CHECK-LABEL: {{sse41_blend_noinline.*closure.*:}}
+        // CHECK-NOT: _mm_blend_ps
+        // CHECK: blendps
+        // CHECK-NOT: _mm_blend_ps
+        // CHECK: ret
+        #[inline(never)] |x, y| unsafe {
+            _mm_blend_ps(x, y, 0b0101)
+        }
+    };
+    f(x, y)
+}
+
+#[no_mangle]
+#[target_feature(enable = "sse4.1")]
+pub fn sse41_blend_doinline(x: __m128, y: __m128) -> __m128 {
+    // check that the closure and _mm_blend_ps are being inlined into the function
+    // CHECK-LABEL: sse41_blend_doinline:
+    // CHECK-NOT: {{sse41_blend_doinline.*closure.*}}
+    // CHECK-NOT: _mm_blend_ps
+    // CHECK: blendps
+    // CHECK-NOT: {{sse41_blend_doinline.*closure.*}}
+    // CHECK-NOT: _mm_blend_ps
+    // CHECK: ret
+    let f = {
+        #[inline] |x, y| unsafe {
+            _mm_blend_ps(x, y, 0b0101)
+        }
+    };
+    f(x, y)
+}

